Source/Data/01 - openXDA.sql

```diff
@@ -3905,6 +3936,72 @@ CREATE TABLE AlarmLog
 )
 GO
 
+CREATE TABLE LatestAlarmLog
+(
+    ID INT IDENTITY(1, 1) NOT NULL PRIMARY KEY,
+    AlarmID INT NOT NULL UNIQUE REFERENCES Alarm(ID),
+    AlarmLogID INT NOT NULL REFERENCES AlarmLog(ID),
+    StartTime DATETIME NOT NULL,
+    EndTime DATETIME NULL
+)
+GO
+
+CREATE NONCLUSTERED INDEX IX_LatestAlarmLog_AlarmLogID
+ON LatestAlarmLog(AlarmLogID ASC)
+GO
+
+CREATE TRIGGER AlarmLog_UpdateLatestAlarmLog
+    ON AlarmLog
+    AFTER INSERT
```

Comment:
Do we need an `UPDATE` trigger too? If `EndTime` gets updated we may need to make sure the update makes it into `LatestAlarmLog`

Reply:
It looks like we don't ever update `AlarmLog` records. But it does seem like something you'd expect should happen from time to time. I guess the range defined by the start/end of the alarm is limited to the range of data in the file we're processing. We don't join the file's data with the archived data to expand the range of an existing alarm or anything like that.